frontend/src/components/workspaces/Templates/TemplatesTable/TemplatesTable.tsx

```diff
@@ -0,0 +1,155 @@
+import { FC } from 'react';
+import { Col, Table } from 'antd';
+import { TemplatesTableRow } from '../TemplatesTableRow';
+import { DownOutlined, RightOutlined } from '@ant-design/icons';
+import { useState } from 'react';
+
+export interface ITemplatesTableProps {
+  templates: Array<{
+    id: string;
+    name: string;
+    gui: boolean;
+    instances: Array<{
+      id: number;
+      name: string;
+      ip: string;
+      status: boolean;
+    }>;
+  }>;
+  role: string;
+  createInstance: (id: string) => void;
+  destroyInstance: (idInstance: number, idTemplate: string) => void;
+  editTemplate: (id: string) => void;
+  deleteTemplate: (id: string) => void;
+}
+
+const TemplatesTable: FC<ITemplatesTableProps> = ({ ...props }) => {
+  const {
+    templates,
+    role,
+    createInstance,
+    editTemplate,
+    deleteTemplate,
+  } = props;
+
+  /**
+   * Our Table has just one column which render all rows using a component TemplateTableRow
+   */
+  /* eslint-disable */
+  const columns = [
+    {
+      title: 'Template',
+      key: 'template',
+      render: (record: {
+        id: string;
+        name: string;
+        gui: boolean;
+        instances: Array<{
+          id: number;
+          name: string;
+          ip: string;
+          status: boolean;
+        }>;
+      }) => (
+        <TemplatesTableRow
+          id={record.id}
+          name={record.name}
+          gui={record.gui}
+          role={role}
+          activeInstances={record.instances ? record.instances.length : 0}
+          createInstance={createInstance}
+          editTemplate={editTemplate}
+          deleteTemplate={deleteTemplate}
+        />
+      ),
+    },
+  ];
+  /* eslint-enable */
+
+  /**
+   * Handle to manage accordion effect, it's possible to allow multiple row expansion (no accordion effect) by switching
+   * handleAccordion() function with the following code
+    const handleAccordion = (
+      expanded: boolean,
+      record: {
+        id: string;
+        name: string;
+        gui: boolean;
+        instances: Array<{
+          id: number;
+          name: string;
+          ip: string;
+          status: boolean;
+        }>;
+      }
+    ) => {
+      expanded
+        ? setExpanedId(expandedId => {
+            expandedId.push(record.id);
+            return expandedId;
+          })
+        : setExpanedId(expandedId => {
+            return expandedId.filter(id => id !== record.id);
+          });
+    };
+   */
+  const [expandedId, setExpandedId] = useState(['']);
+
+  const handleAccordion = (
```

Comment:
I like that you made a function. If this will be needed in other parts of the app (e.g. the activePage), it could useful/interesting to make it a common HOC.

Reply:
Changed applied [here](https://github.com/netgroup-polito/CrownLabs/pull/460/commits/77c8d7a732e466080c9e305ba922259e2ecdaa42)